### PR TITLE
Ball pick up subsystem

### LINKED ISSUE #14: Ball Pickup subsystem creation
Copy feeder code and refactor it for pick up

### PR DESCRIPTION
**Pull request by realtrev**
**Issue:** Ball pickup subsystem creation + buttons

**Files created...**
BallPickUpOn.java: Copy of FeederOn
BallPickUpOff.java: Copy of FeederOff
BallPickUp.java: Copy of Feeder

**Files changed...**
RobotContainer.java: added new variable for operator controller and used operator controller for enabling and disabling ball pickup with A (on) and B (off) buttons.

### CODE REVIEW

**Review by LavaWaffle — COMMENTED** [review]
Code changes look good, but in RobotContainer.java maybe move the joystick import (line 29) next to the other imports

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -21,9 +21,12 @@
 // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=IMPORTS
 import edu.wpi.first.wpilibj2.command.Command;
 
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/RobotContainer.java]: Extra newline

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -21,9 +21,12 @@
...
 // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=IMPORTS
 
 
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/RobotContainer.java]: 2 extra newlines

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -21,9 +21,12 @@
...
+import edu.wpi.first.wpilibj2.command.button.JoystickButton;
```
Comment (bjmcternan) [inline src/main/java/frc/robot/RobotContainer.java]: This import should be up next to the other imports to be clearer

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -38,9 +41,12 @@
 // The robot's subsystems
     public final Drivetrain m_drivetrain = new Drivetrain();
     public final Launcher m_launcher = new Launcher();
+    public final BallPickUp m_ballPickUp = new BallPickUp();
     
 // Joysticks
     public final XboxController m_xboxController = new XboxController(0); 
+    public final XboxController m_operatorXboxController = new XboxController(1); 
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/RobotContainer.java:45]: Extra newline

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -96,6 +102,12 @@ private void configureButtonBindings() {
         // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=BUTTONS
 // Create some buttons
 
+    final JoystickButton ballPickUpOnButton = new JoystickButton(m_operatorXboxController, XboxController.Button.kA.value);
+    ballPickUpOnButton.whenPressed(new BallPickUpOn(m_ballPickUp));
+
+    final JoystickButton ballPickUpOffButton = new JoystickButton(m_operatorXboxController, XboxController.Button.kB.value);
+    ballPickUpOffButton.whenPressed(new BallPickUpOff(m_ballPickUp));
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/RobotContainer.java]: Extra newlines

File: src/main/java/frc/robot/commands/BallPickUpOn.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ */
+public class BallPickUpOn extends CommandBase 
+{
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=VARIABLE_DECLARATIONS
+    private final BallPickUp m_ballPickUp;
+    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=VARIABLE_DECLARATIONS
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+
+    public BallPickUpOn(BallPickUp subsystem){
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/BallPickUpOn.java]: subsystem is a bit generic for a variable name. We should call this ballpickup instead: `BallPickUp ballPickUp`

File: src/main/java/frc/robot/subsystems/BallPickUp.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    // Feeder subsystem
+    public BallPickUp() {
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+        m_ballPickUpMotor = new WPI_TalonFX(6); // TBD
+
+        m_ballPickUpMotor.clearStickyFaults();
+        m_ballPickUpMotor.configFactoryDefault();
+        m_ballPickUpMotor.setInverted(false);
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/BallPickUp.java:42]: Extra newline

File: src/main/java/frc/robot/commands/BallPickUpOff.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ */
+public class BallPickUpOff extends CommandBase 
+{
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=VARIABLE_DECLARATIONS
+    private final BallPickUp m_ballPickUp;
+    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=VARIABLE_DECLARATIONS
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+
+    public BallPickUpOff(BallPickUp subsystem){
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/BallPickUpOff.java]: Let's call `subsystem` `ballPickUp` instead to be more descriptive

**Review by bjmcternan — CHANGES_REQUESTED** [review]

**Review by bjmcternan — APPROVED** [review]

**Pull request merged**